lib/JSON/Schema.pm6

```diff
@@ -375,9 +375,13 @@ class JSON::Schema {
     my class RequiredCheck does Check {
         has Str @.prop;
         method check($value --> Nil) {
-            if $value ~~ Associative && not [&&] $value{@!prop}.map(*.defined) {
-                die X::JSON::Schema::Failed.new:
-                    :$!path, :reason("Object does not have required property");
+            if $value ~~ Associative {
+                for @.prop -> $mandatory_prop {
```

Comment:
Please replace `@.prop` with `@!prop`, it'll be slightly faster this way(in ideal world it'd be optimized by the compiler to be the same, but not right now).

Reply:
@Altai-man By this point the JIT does pretty good on inlining accessors, so they come out pretty alright. :-)

Reply:
Thanks, glad to update my knowledge.

Reply:
For readability I believe @.prop to be better. That's how it is is declared a few lines earlier.

Reply:
When reading a usage of `@.prop`, since it's actually a method call, it's possible that any subclass of the current class might override the accessor method. By contrast, when reading a usage of `@!prop`, you know it's referring unambiguously to the bit of storage declared in this class. Thus I'd argue `@!prop` is more readable in that you can read it and know precisely what it will do, while with `@.prop` there's potential polymorphism in there.